Cabal build does not actually use the library search paths that Autoconf has found

EDIT: Autoconf successfully determines the necessary `-L` search paths, but Cabal does not actually use them. See https://github.com/xmonad/X11/issues/53#issuecomment-525567241 for details.


error:

````
    /usr/bin/ld: cannot find -lXss
````

full log below

````
[...]
````

Using Stackage [LTS 6.35](https://www.stackage.org/lts-6.35) which points at X11 v1.6.1.2 everything works just fine and X11 successfully compiles. Using the newer Stackage LTS 9.0 which points at X11 v1.8, I get the above error and cannot complete the compile.

--- a/X11.buildinfo.in
+++ b/X11.buildinfo.in
@@ -5,3 +5,4 @@ buildable: @BUILD_PACKAGE_BOOL@
 cc-options: @X_CFLAGS@ @CPPFLAGS@
 ld-options: @X_LIBS@ @LDFLAGS@
 extra-libraries: @EXTRA_LIBRARIES@
+extra-lib-dirs: @X_LIBRARIES@

--- a/configure.ac
+++ b/configure.ac
@@ -135,5 +135,8 @@ fi
 
 AC_SUBST([EXTRA_LIBRARIES])
 
+X_LIBRARIES="$x_libraries"
+AC_SUBST([X_LIBRARIES])
+
 AC_CONFIG_FILES([config.mk X11.buildinfo])
 AC_OUTPUT

